### PR TITLE
Added the wal_sync configuration option

### PR DESCRIPTION
**Pull request by monadbobo**
summary:

This option is mainly used to control whether fsync needs to be called every time
after the WAL of the raft writes data(default was false).

### CODE REVIEW

**Comment by monadbobo** [comment]
Jenkins go.

**Comment by monadbobo** [comment]
Jenkins go.

File: src/common/fs/FileUtils.cpp [hunk 1]
```diff
@@ -530,6 +530,13 @@ void FileUtils::Iterator::openFileOrDirectory() {
     status_ = Status::OK();
 }
 
+int FileUtils::sync(int fd) {
+#ifdef HAVE_FDATASYNC
+    return fdatasync(fd);
```
Comment (dangleptr) [inline src/common/fs/FileUtils.cpp]: Be careful!  We should NOT use "fdatasync" here.

Comment (monadbobo) [reply src/common/fs/FileUtils.cpp]: ok, I will fix this.

File: src/kvstore/wal/FileBasedWal.cpp [hunk 2]
```diff
@@ -280,9 +280,17 @@ void FileBasedWal::closeCurrFile() {
...
-    CHECK_EQ(fsync(currFd_), 0) << strerror(errno);
+    if (lastLogId_ > firstLogId_) {
+        if (::fsync(currFd_) == -1) {
+            LOG(WARNING) << "sync wal \"" << currInfo_->path()
+                         << "\" failed, error: " << strerror(errno);
+        }
+    }
     // Close the file
-    CHECK_EQ(close(currFd_), 0) << strerror(errno);
+    if (::close(currFd_) == -1) {
```
Comment (critical27) [inline src/kvstore/wal/FileBasedWal.cpp]: Maybe we can discard CHECK of `utime` below as well. WDYT?

Comment (monadbobo) [reply src/kvstore/wal/FileBasedWal.cpp]: ok, i will fix this.

File: src/kvstore/wal/FileBasedWal.cpp [hunk 3]
```diff
@@ -283,20 +283,33 @@ void FileBasedWal::closeCurrFile() {
         return;
     }
 
-    CHECK_EQ(fsync(currFd_), 0) << strerror(errno);
+    bool synced = false;
+    if (lastLogId_ > firstLogId_) {
```
Comment (critical27) [inline src/kvstore/wal/FileBasedWal.cpp:287]: Why not fsync every time?
BTW, I think if fsync failed, maybe crash would be better, WDYT?

**Pull request merged**